### PR TITLE
A KS for identifying a target after segmentation 

### PR DESCRIPTION
**Pull request by kashefy**
This adds a KS that mimics the IdentityKS except that it applies the mask generated by SegmentationKS to the AFE Signals before inference.

A field is added to the SegmentationHypothesis for storing the hop Size of the Segmentation mask for each source. This is needed to determine the temporal resolution of the mask in case it needs to be re-sampled to match the temporal resolution of the AFE Signal.

Spectral features are excluded from masking.

Dependent on TWOEARS/auditory-front-end#4

### CODE REVIEW

File: src/knowledge_sources/SegmentationKS.m [hunk 1]
```diff
@@ -178,7 +180,7 @@
             % Execute KS if a sufficient amount of data for one block has
             % been gathered
             bExecute = (obj.blackboard.currentSoundTimeIdx - ...
-                obj.lastExecutionTime_s) >= obj.blockSize;
+                obj.lastExecutionTime_s) >= 0.1;
```
Comment (kashefy) [inline src/knowledge_sources/SegmentationKS.m:183]: should we revert this line?

**Comment by hagenw** [comment]
Cool, could you please also make a branch `identify_segment` in TWOEARS/documentation and add a short section describing the new knowledge source under the general section [Identification knowledge sources](https://github.com/TWOEARS/documentation/blob/master/blackboard/knowledge-sources.txt#L294)

**Comment by kashefy** [comment]
@hagenw documentation done (TWOEARS/documentation#8), thanks for pointing it out.

File: src/knowledge_sources/SegmentationKS.m (hunk 1 above)
Comment (kashefy) [inline src/knowledge_sources/SegmentationKS.m:183]: @chrschy, I lowered the threshold to allow more frequent executions of the `SegmentationKS`. The orignal threshold was blockSize, which meant that the segmentation was being done on non-overlapping blocks.
Do you think we should add a parameter to set this threshold at instantiation time?

**Comment by hagenw** [comment]
The documentation is now included, see http://twoears.aipa.tu-berlin.de/doc/latest/blackboard/knowledge-sources/#segment-identity-knowledge-source-segmentidentityks

In order to see how this new knowledge source is working it would also be cool if you could add a new folder containing an example scene to TWOEARS/examples, so everyone can test it.

Otherwise @chrschy should look at your comment above and after that it should be ready for merging.

**Pull request merged**